features/192-asset-gear/index.md

```diff
@@ -0,0 +1,495 @@
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "asset_id": { "type": "string" }
+        },
+        "required": ["asset_id"]
+      }
+    }
+  },
+  "required": ["assets"]
+}
+```
+
+- `asset_id`: The asset ID.
+
+##### Request Example
+
+```json
+{
+  "assets": [
+    { "asset_id": "myimage.png" }
+  ]
+}
+```
+
+#### Response JSON Schema
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "assets": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "asset_id": { "type": "string" },
+          "url": { "type": "string" }
+        },
+        "required": ["asset_id", "url"]
+      }
+    }
+  },
+  "required": ["assets"]
+}
+```
+
+- `url`: The signed URL.
+
+##### Response Example
+
+```json
+{
+  "assets": [
+    {
+      "asset_id": "myimage.png",
+      "url": "https://myappname.skygearapps.com/_asset/get/myimage.png?..."
+    }
+  ]
+}
+```
+
+#### Specification
+
+1. Sign each `asset_id` in `assets`.
+1. Return the result in the original order.
+
+### GET /_asset/get/<asset_id>
+
+#### Request Query String
+
+The asset is eligible for image processing if all of the following hold
+
+- The asset is less than or equal to 20MiB.
+- The detected file type using libmagic is png, jpeg or webp.
+
+#### Specification
+
+1. Reverse proxy the request to the origin server.
+1. Process the asset if the image processing query string is present and the asset is eligible for image processing.
+1. Set `Cache-Control`, `Vary`, `Expires` headers correctly.
+
+## Image Processing
+
+### Image Processing Pipeline via Query String
+
+The query parameter `pipeline` refers to the processing pipeline.
+
+The value follows this syntax.
+
+```
+PIPELINE = ASSET-TYPE ( "/" OPERATION [ "," PARAMETER ]* )+
+PARAMETER = NAME | NAME "_" VALUE
+```
+
+The only supported `ASSET-TYPE` is `image`.
+
+#### Operations
+
+##### format
+
+Transform the asset into the specified format.
+
+- `jpg`: Transform into JPEG.
+- `png`: Transform into PNG.
+- `webp`: Transform into WebP.
+
+###### Example
+
+```
+pipeline=image/format,jpg
+```
+
+Transform the asset into JPEG.
+
+##### quality
+
+Set the quality if the format is `jpg` or `webp`.
+
+- `Q`: Set the absolute quality. Valid values are 1 to 100.
+
+###### Example
+
+```
+pipeline=image/format,jpg/quality,Q_85
+```
+
+Transform the asset into JPEG of quality 0.85.
+
+###### resize
+
+- `m`: Specify the scaling mode. Valid values are `lfit`, `mfit`, `fill`, `pad` and `fixed`.
+- `w`: Specify the target width. Valid values are 1 to 4096.
+- `h`: Specify the target height. Valid values are 1 to 4096.
+- `l`: Specify the longer side of the target. Valid values are 1 to 4096.
+- `s`: Specify the shorter side of the target. Valid values are 1 to 4096.
+- `color`: Specify the fill color when `m` is `pad`.
+
+For the details on the scaling mode, See https://www.alibabacloud.com/help/doc-detail/44688.htm
```

Comment:
We should copy the supported scaling mode into the spec; we cannot ask the implementation to keep matching supported scaling modes with an external service documentation.